src/model/types/account_id/caip10.rs

```diff
@@ -0,0 +1,57 @@
+use {
+    super::eip155::{validate_eip155, Eip155Error, NAMESPACE_EIP155},
+    once_cell::sync::Lazy,
+    regex::Regex,
+    thiserror::Error,
+};
+
+#[derive(Debug, PartialEq, Eq, Error)]
+pub enum Caip10Error {
+    #[error("Account ID is is not a valid CAIP-10 account ID")]
+    Invalid,
+
+    #[error("Account ID uses an unsupported namespace")]
+    UnsupportedNamespace,
+
+    #[error("Account ID is eip155 namespace but: {0}")]
+    Eip155(#[from] Eip155Error),
+}
+
+// https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-10.md#syntax
+static PATTERN: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^([-a-z0-9]{3,8}):([-_a-zA-Z0-9]{1,32}):([-.%a-zA-Z0-9]{1,128})$").unwrap()
```

Comment:
is it safe to unwrap here?

Reply:
Hmm I suppose this only evaluates on first call, not startup. However failures in parsing the regex should all be caught by test cases. Will change to `.expect()` with that explanation

Reply:
Explained in d44a05581fe064d084da842dcf3715fd8ac855a3